gh-actions-workshop/presentation.md

```diff
@@ -0,0 +1,522 @@
+
+```yaml
+jobs:
+  job1:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "${{ format('GitHub SHA: {0}', github.sha) }}"
+      # Example: it will output 'GitHub SHA: 80910396b2df0f055442575dc109f856edfbc24b'
+```
+
+---
+
+### GitHub Actions - Variables
+
+You can set variables either at runtime or at the start of the workflow, that might be useful to conditionally branch the _workflow_ or to get information from APIs:
+
+```yaml
+# defining variables in a static way
+env:
+  DAY_OF_WEEK: Wednesday
+jobs:
+  job1:
+    runs-on: ubuntu-latest
+    env:
+      JOB_ENV: Hello world!
+    steps:
+      - run: echo "$JOB_ENV It's $DAY_OF_WEEK"
+      # It will output  'Hello world! It's Wednesday'
+      - run: echo "${{ env.JOB_ENV }} It's ${{ env.DAY_OF_WEEK }}"
+      # You can also use contexts to access them. I personally prefer contexts to distinguish between
+      #  my variables and shell environment variables
+```
+
+---
+
+### GitHub Actions - Setting dynamic variables
+
+As said previously, it might be useful to set variables at runtime (for example, getting certain data from an API), by appending to `$GITHUB_ENV`:
+
+```yaml
+jobs:
+  job1:
+    run-on: ubuntu-latest
+    steps:
+      - run: echo "JOB_ENV='Hello world!'" >> $GITHUB_ENV
+      - run: echo "$JOB_ENV"
+      # it will output 'Hello world!"
+```
+
+**NOTE:** `JOB_ENV` will only be accessible inside the job `job1`
+
+---
+
+### GitHub Actions - Matrix Strategy
+
+Imagine you need to run tests on all your supported platforms (e.g.: `windows`, `linux`, and `macos`). It's not very productive to duplicate the _workflow_ file 3 times therefore you can use matrices to create _workflow runs_ on a combination of variables:
+
+```yaml
+jobs:
+  example_matrix:
+    strategy:
+      matrix:
+        version: [10, 12, 14]
+        os: [ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.version }}
+```
+
+The matrix strategy will create **all combinations** of the variables (in this case this run will have 6 different configurations). This has a maximum of 256 combinations.
+
+---
+
+### GitHub Actions - Secrets
+
+_Secrets_ allow you to have _workflows_ that can handle **sensitive** information (e.g.: API keys, private keys, etc...). You can only read secrets that you **explicitly** include. You should avoid passing _secrets_ between processes and jobs in your _workflow_ because you might be vulnerable to a **supply-chain attack** (like the xz incident). 
+
+You should pass secrets using the **with** keyword in Actions or printing them on **STDIN** of a process. If you wish to pass it as an argument you must do it indirectly:
+
+```yaml
+
+jobs:
+  job1:
+    env:
+      SUPER_DUPER_SECRET_KEY: "${{ secrets.SUPER_DUPER_SECRET_KEY }}"
+    steps:
+      - run: |
+        echo "$SUPER_DUPER_SECRET_KEY" | some-process
```

Comment:
People may be confused with `some-process`. It is really necessary?

Reply:
yes to demonstrate the most secure way to pipe a secret in